src/base/emu-i386/simx86/cpu-emu.c

```diff
@@ -502,12 +502,12 @@ static void Reg2Cpu (int mode)
 
  /* From now on we'll work on the cpuemu eflags (BUT vm86s eflags can be
   * changed asynchronously by signals) */
-  TheCPU.eflags = vm86s.regs.eflags & SAFE_MASK;
+  TheCPU.eflags = vm86s.regs.eflags & (SAFE_MASK | EFLAGS_VIF);
```

Comment:
Why not just change SAFE_MASK?

Reply:
Just to keep it consistent with kvm.c (see here https://github.com/dosemu2/dosemu2/blob/966378828a6e3debeed00f8cc6a96964b50ce0bc/src/base/emu-i386/kvm.c#L811) SAFE_MASK is also used for popf where it should not include VIF